cpu/cc2538/periph/uart.c

```diff
@@ -161,26 +115,12 @@ int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
     }
 
     /* register callbacks */
-    uart_config[uart].rx_cb = rx_cb;
-    uart_config[uart].arg = arg;
+    uart_ctx[uart].rx_cb = rx_cb;
+    uart_ctx[uart].arg = arg;
 
     /* configure interrupts and enable RX interrupt */
-    switch (uart) {
-#if UART_0_EN
-        case UART_0:
-            NVIC_SetPriority(UART0_IRQn, UART_IRQ_PRIO);
-            NVIC_EnableIRQ(UART0_IRQn);
-            break;
-#endif
-#if UART_1_EN
-        case UART_1:
-            NVIC_SetPriority(UART1_IRQn, UART_IRQ_PRIO);
-            NVIC_EnableIRQ(UART1_IRQn);
-            break;
-#endif
-        default:
-            return UART_NODEV;
-    }
+    NVIC_SetPriority(uart_config[uart].irq, UART_IRQ_PRIO);
```

Comment:
drop this line. No need for setting the priority, as this is done globally for all interrupts in the cortexm initialization. (also some ancient leftover...)